src/QirRuntime/lib/Tracer/README.md

```diff
@@ -0,0 +1,177 @@
+# Resource Tracer Design Document #
+
+The purpose of the Resource Tracer is to provide efficient and flexible way to estimate resources of a quantum program
+ in QIR representation. The estimates are calculated by simulating execution of the program (as opposed to the static
+ analysis).
+
+In addition to the standard QIR runtime functions, the quantum program will have to:

```

Comment:
> **In addition to the** ... **runtime functions**, the **quantum program** will have to...

a) The beginning of the sentence sounds incomplete. In addition to **doing what** with runtime functions? In addition to ... **calling? converting?** ... the runtime functions.
b)

Reply:
Reworded the lead sentence. The "quantum program" here means the package, consisting of QIR and C++ code. It's not Q# specific and it's up to the clients how they create such a program. I'm open to improving the terminology but I'd like to avoid creating the impression that the tracer depends on Q# compiler (unless we make it so).